lib/auth/auth.go

```diff
@@ -676,7 +676,10 @@ func (a *Server) generateUserCert(req certRequest) (*certs, error) {
 	if err != nil {
 		return nil, trace.Wrap(err)
 	}
-	if req.routeToCluster != "" && clusterName != req.routeToCluster {
+	if req.routeToCluster == "" {
+		req.routeToCluster = clusterName
+	}
```

Comment:
Could you provide some context on why this change is needed? PR description doesn't mention anything.

Reply:
You could use an empty cluster name to bypass RBAC rules for cluster labels.
The leaf cluster will trust certs signed by the root cluster and interpret empty `RouteToCluster` as "this cluster". The leaf cluster doesn't know about cluster label restrictions from roles in the root cluster.